src/App.js

```diff
@@ -7,29 +7,67 @@ class App extends React.Component {
   //#region you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   //#endregion this component is going to take care of state, and any change handlers you need to work with your state
-  constructor() {
-    super();
-    this.state={
+    state={
       goals: [],
-      temp: "",
-      message: ""
+      temp: ""
     };
-  }
+
 
   inputFunction = (event) => {
-    this.setState({temp: event.target.value});//store input in temporary space.
+    this.setState({[event.target.name]: event.target.value});//store input in temporary space.
   }
 
   addTodoFunction = (event) => {
     event.preventDefault();
-    const task  = {
-    name: this.state.temp,
-    id: Date.now(),
-    completed: false
+    if (this.state.temp !== "") {
+    const task = this.state.temp;
+    const id = Date.now();
+    const newTodo = {
+      name: task,
+      id: id,
+      completed: false
+    }
+
+    let todos = this.state.goals; 
+    todos.push(newTodo);
+    this.setState({goals: todos, temp: ""});//
+    }
   }
 
-  this.state.goals.push(task);
-}
+  clearFunction = (event) => {
+    event.preventDefault();
+    this.setState({
+      goals: this.state.goals.filter(goal => goal.completed === false
+        )})
+  }
+
+  removeFunction = (id) => {
```

Comment:
nice to see this all working. the app is really coming together. good clean code and a nice program flow